### PR TITLE
[READY]Aimable wizard lightning bolts

### PR DESCRIPTION
**Pull request by silicons**
Makes a spell that has lightning bolt as a projectile. Wizard is immune to tesla hits while it is active.

### CODE REVIEW

File: code/modules/spells/spell_types/lightning.dm [hunk 1]
```diff
@@ -81,4 +81,64 @@
...
+	if(..())
+		return FALSE
+	if(!cast_check(0, ranged_ability_user))
+		remove_ranged_ability()
+		return FALSE
+	var/list/targets = list(target)
+	perform(targets,user = ranged_ability_user)
+	return TRUE
+
+/obj/effect/proc_holder/spell/fireball/cast(list/targets, mob/living/user)
```
Comment (ChangelingRain) [inline code/modules/spells/spell_types/lightning.dm]: >fireball

Comment (kevinz000) [inline code/modules/spells/spell_types/lightning.dm]: F U C K

File: code/modules/spells/spell_types/lightning.dm [hunk 2]
```diff
@@ -81,4 +81,64 @@
...
+/obj/effect/proc_holder/lightningbolt
+	name = "Lightning Bolt"
+	desc = "Fire a high powered lightning bolt at your foes!"
+	school = "evocation"
+	charge_max = 200
+	clothes_req = 0
+	invocation = "UN'LTD P'WAH"
+	invocation_type = "shout"
+	cooldown_min = 30
+	var/projectile_type = /obj/item/projectile/magic/lightning
```
Comment (ChangelingRain) [inline code/modules/spells/spell_types/lightning.dm]: you're doing something wrong here, given that this and fireball are not both subtypes of a type with this

Comment (kevinz000) [inline code/modules/spells/spell_types/lightning.dm]: hehheh

File: code/modules/projectiles/projectile/magic.dm [hunk 3]
```diff
@@ -387,3 +340,85 @@
...
+	damage_type = BRUTE
+	nodamage = 0
+
+	//explosion values
+	var/exp_heavy = 0
+	var/exp_light = 2
+	var/exp_flash = 3
+	var/exp_fire = 2
+
+/obj/item/projectile/magic/fireball/aoe/on_hit(target)
```
Comment (ChangelingRain) [inline code/modules/projectiles/projectile/magic.dm]: type issues!

File: code/modules/projectiles/projectile/magic.dm [hunk 4]
```diff
@@ -387,3 +340,85 @@
...
+	nodamage = 0
+	speed = 0.3
+	flag = "magic"
+
+	var/tesla_power = 20000
+	var/tesla_range = 15
+	var/chain
+	var/mob/living/caster
+
+/obj/item/projectile/magic/lightning/aoe/fire(setAngle)
```
Comment (ChangelingRain) [inline code/modules/projectiles/projectile/magic.dm]: type issues!

File: code/modules/spells/spell_types/lightning.dm [hunk 5]
```diff
@@ -81,4 +81,64 @@
...
+/obj/effect/proc_holder/spell/lightningbolt
+	name = "Lightning Bolt"
+	desc = "Fire a high powered lightning bolt at your foes!"
+	school = "evocation"
+	charge_max = 200
+	clothes_req = 1
+	invocation = "UN'LTD P'WAH"
+	invocation_type = "shout"
+	cooldown_min = 30
+	var/projectile_type = /obj/item/projectile/magic/lightning
```
Comment (ChangelingRain) [inline code/modules/spells/spell_types/lightning.dm]: you're still doing something wrong here, given that this and fireball are not both subtypes of a type with this var
previous comment was unclear

File: code/modules/spells/spell_types/lightning.dm [hunk 6]
```diff
@@ -81,4 +81,64 @@
...
+/obj/effect/proc_holder/spell/lightningbolt
+	name = "Lightning Bolt"
+	desc = "Fire a high powered lightning bolt at your foes!"
+	school = "evocation"
+	charge_max = 200
+	clothes_req = 1
+	invocation = "UN'LTD P'WAH"
+	invocation_type = "shout"
+	cooldown_min = 30
+	var/projectile_type = /obj/item/projectile/magic/aoe/lightning
```
Comment (ChangelingRain) [inline code/modules/spells/spell_types/lightning.dm]: you're _still_ doing something wrong here, given that this and the fireball spell are not both subtypes of a type with this var

Comment (kevinz000) [inline code/modules/spells/spell_types/lightning.dm]: .. woooot?!

Comment (kevinz000) [inline code/modules/spells/spell_types/lightning.dm]: OH
THE SPELLS
FUCK.

Comment (ChangelingRain) [inline code/modules/spells/spell_types/lightning.dm]: you copied this from the fireball spell.
it should not be copied; this at minimum should be a subtype of the fireball spell so this and cast() aren't copied from fireball.

**Comment by PKPenguin321** [comment]
the odd third child of fireball and the lightning spell

**Comment by Incoming5643** [comment]
as much as it pains me to say this, you might have to blacklist this bolt from the staff of chaos, since any time it was rolled it'd probably shock the staff holder.

**Comment by ChangelingRain** [comment]
you'll also have to blacklist the base aoe projectile

**Comment by Incoming5643** [comment]
actually while you're there blacklist the hellish fireball too, it was essentially giving fireball double odds of staff of chaos

**Comment by XDTM** [comment]
Magicarps and chaos carps also pick magic bolts randomly.

**Pull request merged**